docs-en/chapter_stack_and_queue/queue.md

```diff
@@ -1,14 +1,14 @@
 # Queue
 
-"Queue" is a linear data structure that follows the First-In-First-Out (FIFO) rule. As the name suggests, a queue simulates the phenomenon of lining up, where newcomers join the back of the queue, and people at the front of the queue leave one by one.
+"Queue" is a linear data structure that follows the First-In-First-Out (FIFO) rule. As the name suggests, a queue simulates the phenomenon of lining up, where newcomers join at the rear of the queue, and people at the front leave one by one.
 
-As shown in the figure below, we call the front of the queue the "head" and the back the "tail." The operation of adding elements to the tail of the queue is termed "enqueue," and the operation of removing elements from the head is termed "dequeue."
+As shown in the figure below, we call the front of the queue the "head" and the back the "tail." The operation of adding elements to the rear of the queue is termed "enqueue," and the operation of removing elements from the front is termed "dequeue."
 
 ![Queue's First-In-First-Out Rule](queue.assets/queue_operations.png)
 
 ## Common Operations on Queue
 
```

Comment:
> I would prefer using "diagram" instead of "figure". @krahets what do you think? If we agree on all visual demonstration of algorithm should be called "diagram", we should apply this to all places.

I agree that using "diagram" could be beneficial for clarity and precision. 

However, "figure" is more general and is used widely in academic papers and books. All the "figure below/above" will be replaced by "Figure xx" during the build process, as shown in the screenshot below:

<img width="856" alt="image" src="https://github.com/krahets/hello-algo/assets/26993056/6e0f0b52-e90f-4ddf-b860-6ed0b49efd6e">

So I would recommend using "figure" throughout the book.

Reply:
Cool, that makes sense.